ledger/txtail.go

```diff
@@ -154,18 +229,81 @@ func (t *txTail) committedUpTo(rnd basics.Round) (retRound, lookback basics.Roun
 		delete(t.lastValid, t.lowWaterMark)
 	}
```

Comment:
```
for t.lowWaterMark < rnd {
  t.lowWaterMark++
  delete(t.lastValid, t.lowWaterMark)
}
```

Reply:
no need to modify code only because of a style preferences

Reply:
the code I'm suggesting is not equivalent